JASP-Engine/JASP/R/commonAnovaBayesian.R

```diff
@@ -1834,11 +1964,43 @@
     # we're doing model averaged inference
     
     # sample from the BMA posterior
-    samples <- matrix(NA, nrow = nIter, ncol = ncol(posterior))
-    for (i in seq_len(ncol(posterior))) {
-      samples[, i] <- .BANOVAreSample(n = nIter, x = posterior[, i, "x"], y = posterior[, i, "y"], prop0 = prop0[i])
-    }
+    effects <- model[["effects"]]
+    rownames(effects) <- NULL # drop rownames, otherwise they get copied later on
+    postProbs <- model[["postProbs"]]
+    postProbs[is.na(postProbs)] <- 0
+    modelIdx <- sample(nrow(effects), nIter, TRUE, postProbs) # resample the models according to posterior prob
+    samples <- matrix(0, nrow = nIter, ncol = ncol(posterior))
     
+    # get for each parameter which variable they belong to (e.g., 0 and 1 belong to contBinom)
+    stems <- sapply(strsplit(colnames(posterior), "-"), `[`, 1L) # before - is the variable name
+    bases <- c("mu", names(levelInfo[["levelNames"]])) # all options
+    pos <- match(stems, bases) # lookup parameters in variable names
+  
+    # sort the terms inside bases and colnames(effects) so that they match
+    # NOTE: this cannot be done when creating effects because then the colnames don't match with the formulas...
+    bases <- sapply(strsplit(bases, ":"), function(x) paste0(sort(x), collapse = ":"))
```

Comment:
Are these `bases` base64 encoded? Otherwise rogue ":" would be a problem

Reply:
These are base64 as long as the formulas and data set given to BayesFactor are.